### PR TITLE
Feature/pipline infra setup

### PR DESCRIPTION
**Pull request by wencfy**
- Build the CD pipeline.
- Behavioral Driven Development (BDD) Testing.
- Changed behavior test browser to chrome.
- Added compatible logics for converting - format date to / format date for chrome.

Resolves: #86 #87 #90

### CODE REVIEW

**Comment by wencfy** [comment]
Finally successfully deployed!

- Trigger url: https://cd-pipeline-wen-wcfy-dev.apps.sandbox-m4.g2pi.p1.openshiftapps.com
- Deployed website url: https://promotions-wen-wcfy-dev.apps.sandbox-m4.g2pi.p1.openshiftapps.com

Changes: .tekton & k8s

**Comment by wencfy** [comment]
Done behave test task.

- Changed the browser to chrome.
- Added compatible logics for converting - format date to / format date for chrome in web_steps.

image

File: .tekton/tasks.yaml [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+      type: array
+      default: []
+    - name: secret-name
+      description: The name of the secret containing a database_uri key
+      type: string
+      default: "postgres-creds"
+    - name: secret-key
+      description: The name of the key that contains the database uri
+      type: string
+      default: "database_uri"
```
Comment (yc6714) [inline .tekton/tasks.yaml:101]: please change here to `database-uri`

**Review by yc6714 — APPROVED** [review]
LGTM Good Job Wen

**Review by pennyyang17 — APPROVED** [review]
Thanks Wen!!

**Review by raghavm1 — APPROVED** [review]
Thanks, Wen!

**Review by ian-Liaozy — APPROVED** [review]
Thanks @wencfy !

**Pull request merged**